### PR TITLE
Feature/1577

### PR DESCRIPTION
**Pull request by NSkelsey**
This addresses #1577 which gives each recipient the ability to toggle notifications for individual submissions on and off.

It uses the extensibility of the `tip.setVar( )` function on the client side and the `rtip put` handler on the server to get the job done. Finally, the actual check on whether or not to actually send the mail or not occurs in `process_mail_creation`.

I believe that `process_mail_creation` needs a bit of scrutiny. The function already does many different things and while it looked like the place to put the logic. It seems like there could be a better place to intercept the mail spooling.

### CODE REVIEW

File: backend/globaleaks/models/__init__.py [hunk 1]
```diff
@@ -315,7 +315,7 @@ class ReceiverTip(Model):
 
     new = Int(default=True)
 
-    enable_notifications = Int(default=True)
+    enable_notifications = Bool(default=True)
```
Comment (NSkelsey) [inline backend/globaleaks/models/__init__.py]: @evilaliv3 please scrutinize.

Comment (evilaliv3) [inline backend/globaleaks/models/__init__.py]: you are right!

File: client/tests/end2end/test-globaleaks-process.js [hunk 2]
```diff
@@ -274,6 +274,23 @@ describe('globaLeaks process', function() {
     }
   });
 
+
+  fit('Recipient should be able to disable and renable email notifications', function() {
```
Comment (evilaliv3) [inline client/tests/end2end/test-globaleaks-process.js]: fit() stays for fitness?

File: client/tests/end2end/test-globaleaks-process.js [hunk 3]
```diff
@@ -274,6 +274,23 @@ describe('globaLeaks process', function() {
...
+    login_receiver(receiver_username, receiver_password);
+    element(by.id('tip-0')).click();
+    var silence = element(by.id('tip-action-silence'));
+    silence.click();
+    var notif = element(by.id('tip-action-notify'));
+    notif.evaluate('tip.enable_notifications').then(function(enabled) {
+      expect(enabled).toEqual(false);
```
Comment (evilaliv3) [inline client/tests/end2end/test-globaleaks-process.js]: i like this fine checks!

File: backend/globaleaks/jobs/notification_sched.py [hunk 4]
```diff
@@ -120,9 +120,15 @@ def process_ReceiverFile(self, store, rfile, data):
         self.process_mail_creation(store, data)
 
     def process_mail_creation(self, store, data):
+        receiver_id = data['receiver']['id']
+
+        # Do not spool emails if the receiver has opted out of ntfns for this tip.
+        if not data['tip']['enable_notifications']:
+          log.debug("Discarding emails for %s due to receiver's preference." % receiver_id)
```
Comment (evilaliv3) [inline backend/globaleaks/jobs/notification_sched.py]: 'receiver' is generally nowadays used only in the code; if the message is for the user(of the webapplication or the admin that looks the logs) the texts should prefer using 'recipient'. various translators  indipendently suggested this.

**Comment by evilaliv3** [comment]
Well done @NSkelsey !

**Pull request merged**